### PR TITLE
Ape Dependencies, Full relock, and CI Integration

### PR DESCRIPTION
**Pull request by derekpierre**
Based over #3108 . 

Relock dependencies to incorporate ape and update CI.

### CODE REVIEW

File: dev-requirements.txt [hunk 1]
```diff
@@ -1,37 +1,133 @@
...
-hypothesis==6.72.1
-identify==2.5.22 ; python_version >= '3.7'
+hexbytes==0.3.0 ; python_version >= '3.7' and python_version < '4'
+hypothesis==6.74.1
+identify==2.5.23 ; python_version >= '3.7'
 idna==3.4 ; python_version >= '3.5'
+ijson==3.2.0.post0
+importlib-metadata==6.6.0 ; python_version >= '3.7'
 iniconfig==2.0.0 ; python_version >= '3.7'
+ipython==8.13.0 ; python_version >= '3.8'
```
Comment (KPrasch) [inline dev-requirements.txt:52]: I'm wondering how this dependency was resolved.  Did you build this from python 3.8.x?


```
...
ImportError: 
IPython 8.13+ supports Python 3.9 and above, following NEP 29.
IPython 8.0-8.12 supports Python 3.8 and above, following NEP 29.
```

File: Pipfile [hunk 2]
```diff
@@ -42,6 +40,8 @@ tabulate = "*"
 # Template
 jinja2 = "<3.1.0"  # because of docs
 
+watchdog = "<3.0.0"  # needed for eth-ape to be happy
```
Comment (KPrasch) [inline Pipfile:43]: ...interesting?  Any idea why?

File: Pipfile [hunk 3]
```diff
@@ -51,11 +51,13 @@ pytest-cov = "*"
...
 pytest-timeout = "*"
 # Tools
+eth-ape = ">=0.6.3,<0.7.0"
+ape-solidity = ">=0.6.0,<0.7.0"
 hypothesis = "*"
 pre-commit = "2.12.1"
 pyflakes = "*"
 coverage = "<=6.5.0"
-py-solc-x = "==0.10.1"
+py-solc-x = "==1.1.1"   # docs version matches this one
```
Comment (KPrasch) [inline Pipfile:60]: this dependency can be removed since it will be managed by eth-ape.

Comment (derekpierre) [reply Pipfile]: Should be gone now.

File: dev-requirements.txt (hunk 1 above)
Comment (derekpierre) [reply dev-requirements.txt]: > Did you relock from python 3.8.x?
I have been using python 3.8.16

**Review by KPrasch — APPROVED** [review]

**Pull request merged**